processing/src/main/java/org/apache/druid/segment/transform/ExpressionTransform.java

```diff
@@ -111,9 +110,7 @@ public Object eval(final Row row)
     public List<String> evalDimension(Row row)
     {
       try {
-        return Rows.objectToStrings(
```

Comment:
note: removal of `coerceEvalToObjectOrList` will have a sideeffect that it will not anymore turn `[singleVal]` into just `singleVal` - I don't know if that's important or not - but I wonder how `array_slice(dim, 0, 1)` is handled before/after the patch; might worth a quick check

Reply:
it is handled correctly in the serializer/deserializer.  Also added test to make sure that we are getting array with 1 value from transform in the single value case.

Reply:
yea, this change is ok in this case because `Rows.objectToStrings` is translating anything into `List<String>`, this method was just previously re-using the method used by the expression dimension selector on the query side to which itself matches https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/segment/DimensionSelector.java#L141, but since in this transform it was wrapped in Rows.objectToStrings, that part of the behavior wasn't important, just the converting an array to a list of strings part.